lib/src/arithmetic/compound_adder.dart

```diff
@@ -123,71 +126,76 @@ class CarrySelectCompoundAdder extends CompoundAdder {
         ? addInput('subtractIn', subtractIn, width: subtractIn.width)
         : null;
     // output bits lists
-    final sumList0 = <Logic>[];
-    final sumList1 = <Logic>[];
+    final sumList0F = <Logic>[];
+    final sumList1F = <Logic>[];
     // carryout of previous adder block
     // for sum and sum+1
-    Logic? carry0;
-    Logic? carry1;
+    Logic carry0 = Const(0);
+    Logic carry1 = Const(0);
     // Get size of each adder block
     final adderSplit = widthGen(a.width);
     // 1st output bit index of each block
     var blockStartIdx = 0;
     for (var i = 0; i < adderSplit.length; ++i) {
       // input width of current adder block
       final blockWidth = adderSplit[i];
+      final sum0Ary = Logic(width: blockWidth);
+      final sum1Ary = Logic(width: blockWidth);
       if (blockWidth <= 0) {
         throw RohdHclException('non-positive adder block size.');
       }
       if (blockWidth + blockStartIdx > a.width) {
         throw RohdHclException('oversized adders sequence.');
       }
-      final blockA = Logic(name: 'block_${i}_a', width: blockWidth);
-      final blockB = Logic(name: 'block_${i}_b', width: blockWidth);
-      blockA <= a.getRange(blockStartIdx, blockStartIdx + blockWidth);
-      blockB <= b.getRange(blockStartIdx, blockStartIdx + blockWidth);
       // Build sub adders for 0 and 1 carryin values
-      final fullAdder0 = adderGen(blockA, blockB,
-          subtractIn: subtractIn, carryIn: Const(0), name: 'block0_$i');
-      final fullAdder1 = adderGen(blockA, blockB,
-          subtractIn: subtractIn, carryIn: Const(1), name: 'block1_$i');
-      for (var bitIdx = 0; bitIdx < blockWidth; ++bitIdx) {
-        if (i == 0) {
-          // connect directly to respective sum output bit
-          sumList0.add(fullAdder0.sum[bitIdx]);
-          sumList1.add(fullAdder1.sum[bitIdx]);
-        } else {
-          final bitOut0 = Logic(name: 'bit0_${blockStartIdx + bitIdx}');
-          final bitOut1 = Logic(name: 'bit1_${blockStartIdx + bitIdx}');
-          // select adder output from adder matching carryin value
-          bitOut0 <=
-              mux(carry0!, fullAdder1.sum[bitIdx], fullAdder0.sum[bitIdx]);
-          bitOut1 <=
-              mux(carry1!, fullAdder1.sum[bitIdx], fullAdder0.sum[bitIdx]);
-          sumList0.add(bitOut0);
-          sumList1.add(bitOut1);
-        }
-      }
+      final fullAdder0 = adderGen(
+          a.getRange(blockStartIdx, blockStartIdx + blockWidth),
+          b.getRange(blockStartIdx, blockStartIdx + blockWidth),
+          subtractIn: subtractIn,
+          carryIn: Const(0),
+          name: 'block0_$i');
+      final fullAdder1 = adderGen(
+          a.getRange(blockStartIdx, blockStartIdx + blockWidth),
+          b.getRange(blockStartIdx, blockStartIdx + blockWidth),
+          subtractIn: subtractIn,
+          carryIn: Const(1),
+          name: 'block1_$i');
+      sum0Ary <=
+          [
```

Comment:
why bit-blast this instead of just using `getRange` to grab the bits?  something like 
```dart
[mux(carry0, fullAdder1.sum.getRange(1, blockWidth), fullAdder0.sum.getRange(1, blockWidth)), fullAdder0.sum[0]].swizzle().named('block_${i}_sum0Ary')
```

synthesizes the same way but simulates simpler, maybe more readable, and also makes nicer SV